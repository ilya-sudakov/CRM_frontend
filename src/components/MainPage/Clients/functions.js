export const clientsFormHeaderMenu = (name) => {
  return [
    { pageTitle: 'История работы', pageName: 'workHistory' },
    {
      pageTitle: `Данные ${name}а`,
      pageName: 'clientData',
    },
  ];
};
