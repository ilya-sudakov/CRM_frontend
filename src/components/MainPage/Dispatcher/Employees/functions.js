import {
  formatDateString,
  getEmployeeNameText,
} from 'Utils/functions.jsx';
import {
  createPDF,
  getPDFTitleObject,
  getInputElementTextPDF,
  defaultStylesPDF,
} from 'Utils/pdfFunctions.js';

export const filterEmployeesBySearchQuery = (data, searchQuery) => {
  const query = searchQuery.toLowerCase();
  return data.filter((item) => {
    if (item.name === null) return false;
    const isFound =
      item.lastName.toLowerCase().includes(query) ||
      item.name.toLowerCase().includes(query) ||
      item.middleName.toLowerCase().includes(query) ||
      item.id.toString().includes(query) ||
      (item.yearOfBirth && item.yearOfBirth.toString().includes(query)) ||
      (item.dateOfBirth && item.dateOfBirth.toString().includes(query)) ||
      item.citizenship.toLowerCase().includes(query) ||
      item.workshop.toLowerCase().includes(query) ||
      item.position.toLowerCase().includes(query) ||
      item.comment.toLowerCase().includes(query) ||
      item.relevance.toLowerCase().includes(query);
    return isFound;
  });
};

const getEmployeesTablePDF = (employeeInfo) => {
  return {
    table: {
      widths: ['*', 70, 80, 120, 100],
      body: [
        [
          { text: 'ФИО', style: 'tableHeader' },
          { text: 'Дата рождения', style: 'tableHeader' },
          { text: 'Гражданство', style: 'tableHeader' },
          { text: 'Должность', style: 'tableHeader' },
          { text: '', style: 'tableHeader' },
        ],
        ...employeeInfo,
      ],
    },
  };
};

export const getEmployeesByWorkshopListPdfText = (employees = [], workshop) => {
  let employeesList = [],
    employeeInfo = [];
  employees.map((item) => {
    employeeInfo.push([
      getEmployeeNameText(item),
      formatDateString(item.yearOfBirth),
      item.citizenship,
      item.position,
      '',
    ]);
  });
  employeesList.push(getEmployeesTablePDF(employeeInfo));
  const dd = {
    info: {
      title: `Список сотрудников - ${workshop}`,
    },
    content: [
      getPDFTitleObject(`Список сотрудников ${workshop}\n`),
      ...employeesList,
    ],
    styles: defaultStylesPDF,
  };
  createPDF(dd);
};

export const getEmployeesListPdfText = (employees, workshops) => {
  const employeesList = [];
  workshops.map((workshop) => {
    employeesList.push(getInputElementTextPDF('Подразделение', workshop));
    let employeeInfo = [];
    employees.map((employee) => {
      if (
        (workshop === employee.workshop && employee.relevance !== 'Уволен') ||
        (workshop === 'Уволенные' && employee.relevance === 'Уволен')
      ) {
        employeeInfo.push([
          getEmployeeNameText(employee),
          formatDateString(employee.yearOfBirth),
          employee.citizenship,
          employee.position,
          '',
        ]);
      }
    });
    employeesList.push(getEmployeesTablePDF(employeeInfo));
  });
  const dd = {
    info: {
      title: 'Список сотрудников',
    },
    content: [getPDFTitleObject(`Список сотрудников`), ...employeesList],
    styles: defaultStylesPDF,
  };
  createPDF(dd);
};
