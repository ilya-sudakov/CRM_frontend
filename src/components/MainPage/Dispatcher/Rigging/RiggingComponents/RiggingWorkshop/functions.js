export const filterRigItems = (data = [], curPage) => {
  return data.filter((item) => {
    if (item.color === 'completed' && curPage === 'Завершено') {
      return true;
    }
    if (curPage === 'Активные' && item.color !== 'completed') {
      return true;
    }
  });
};
