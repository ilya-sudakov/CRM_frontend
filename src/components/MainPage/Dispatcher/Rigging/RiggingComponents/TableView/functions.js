export const isPartHidden = (list, index) => {
  index = Number.parseInt(index);
  let check = true;
  list.map((element) => {
    if (element.id === index) {
      check = element.hidden;
    }
  });
  return check;
};

export const searchQuery = (data, query) => {
  let re = /[.,\_-\s]/gi;
  const newQuery = query.toLowerCase();
  return data.filter(
    (item) =>
      item.id.toString().includes(newQuery) ||
      item.comment.toLowerCase().includes(newQuery) ||
      item.name.toLowerCase().includes(newQuery) ||
      item.number
        .toLowerCase()
        .replace(re, "")
        .includes(newQuery.replace(re, ""))
  );
};

export const sortStampParts = (data) => {
  return data.sort((a, b) => {
    if (a.number < b.number) {
      return -1;
    }
    if (a.number > b.number) {
      return 1;
    }
    if (a.number === b.number && a.id < b.id) {
      return -1;
    }
    if (a.number === b.number && a.id > b.id) {
      return 1;
    }
    return 0;
  });
};
