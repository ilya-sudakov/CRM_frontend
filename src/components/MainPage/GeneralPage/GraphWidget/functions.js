export const getDateFromWeekdayIndex = (dayIndex, weekOffset) => {
  return new Date(
    new Date().setDate(
      new Date().getDate() +
        -7 * weekOffset -
        new Date(
          new Date().setDate(new Date().getDate() + -7 * weekOffset),
        ).getDay() +
        dayIndex,
    ),
  );
};

export const getWeekdaysListWithOffset = (offset = 0) => {
  const curDay = new Date(
    new Date().setDate(new Date().getDate() + -7 * offset),
  );
  let week = [];
  for (
    let i = 1;
    i <=
    (offset === 0 ? (new Date().getDay() === 0 ? 7 : new Date().getDay()) : 7);
    i++
  ) {
    let first = curDay.getDate() - curDay.getDay() + i;
    let day = new Date(curDay.setDate(first));
    week.push(day);
  }
  return week;
};
