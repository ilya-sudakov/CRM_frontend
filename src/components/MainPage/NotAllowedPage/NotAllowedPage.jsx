import { Link } from 'react-router-dom';
import '../ErrorPage/ErrorPage.scss';

const NotAllowedPage = () => {
  return (
    <div className="error-page">
      <div className="error-page__title">
        <span className="error-page__error-code">403</span>
        Доступ на эту страницу запрещен
      </div>
      <div className="error-page__description">
        Вы не имеете доступа к этой странице.
        <br />
        <br />
        Если это ошибка, и запрашиваемая страница должна быть доступна для
        просмотра - напишите об этом в{' '}
        <Link className="error-page__link" to="/feedback">
          обратной связи
        </Link>
      </div>
      <Link className="error-page__button" to="/">
        На главную
      </Link>
    </div>
  );
};

export default NotAllowedPage;
