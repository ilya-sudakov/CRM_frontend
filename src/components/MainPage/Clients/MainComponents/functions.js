import { exportClientsEmailsCSV } from 'Utils/xlsxFunctions.js';
import { getClients } from 'Utils/RequestsAPI/Clients.jsx';

export const sortClients = (clients, searchQuery, sortOrder) => {
  return clients.sort((a, b) => {
    if (searchQuery !== '') {
      const first =
          sortOrder.curSort === 'nextDateContact'
            ? new Date(a[sortOrder.curSort])
            : a[sortOrder.curSort],
        second =
          sortOrder.curSort === 'nextDateContact'
            ? new Date(b[sortOrder.curSort])
            : b[sortOrder.curSort];
      if (first < second) {
        return sortOrder[sortOrder.curSort] === 'desc' ? 1 : -1;
      }
      if (first > second) {
        return sortOrder[sortOrder.curSort] === 'desc' ? -1 : 1;
      }
      return 0;
    }
  });
};

export const getEmailsExcel = () => {
  let totalClients = 1;
  let clients = [];
  return getClients(1)
    .then((res) => res.json())
    .then((res) => {
      totalClients = res.totalElements;
      return getClients(totalClients);
    })
    .then((res) => res.json())
    .then((res) => {
      clients = res.content;
      console.log(clients);
      exportClientsEmailsCSV(clients);
    })
    .catch((error) => {
      console.log(error);
    });
};
