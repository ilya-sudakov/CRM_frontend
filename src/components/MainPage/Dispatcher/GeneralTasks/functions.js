import { formatDateString } from 'Utils/functions.jsx';

export const filterCompletedTasks = (tasks, curPage) => {
  return tasks.filter(
    (task) =>
      (curPage === 'В процессе' && task.condition !== 'Выполнено') ||
      (curPage === 'Завершено' && task.condition === 'Выполнено'),
  );
};

export const filterTasksUsers = (tasks, taskUsers) => {
  return tasks.filter((item) => {
    let check = false;
    let noActiveStatuses = true;
    Object.entries(taskUsers).map((user) => {
      Object.entries(taskUsers).map((user) => {
        if (user[1]) {
          noActiveStatuses = false;
        }
      });
      if (
        noActiveStatuses === true ||
        (user[1] && user[0] === item.responsible)
      ) {
        check = true;
        return;
      }
    });
    return check;
  });
};

export const filterSearchQuery = (data, searchQuery) => {
  const query = searchQuery.toLowerCase();
  return data.filter(
    (item) =>
      item.id.toString().includes(query) ||
      item.description.toLowerCase().includes(query) ||
      item.responsible.toLowerCase().includes(query) ||
      item.status.toLowerCase().includes(query) ||
      formatDateString(item.dateCreated).includes(query) ||
      formatDateString(item.dateControl).includes(query),
  );
};

export const getTasksDefaultInputs = (username = '') => {
  return [
    {
      name: 'dateCreated',
      defaultValue: new Date(),
      isRequired: true,
      isValid: true,
    },
    {
      name: 'dateControl',
      defaultValue: new Date(new Date().setDate(new Date().getDate() + 7)),
      isRequired: true,
      isValid: true,
    },
    {
      name: 'description',
      defaultValue: '',
      isRequired: true,
    },
    {
      name: 'responsible',
      defaultValue: username,
      isRequired: true,
      isValid: true,
    },
    {
      name: 'status',
      defaultValue: '',
    },
    {
      name: 'condition',
      defaultValue: 'В процессе',
    },
  ];
};
