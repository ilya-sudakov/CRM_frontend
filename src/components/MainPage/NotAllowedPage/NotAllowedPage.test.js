import NotAllowedPage from './NotAllowedPage.jsx';
import { cleanup } from '@testing-library/react';
import { renderWithRouter } from '../../../utils/testing/functions.js';

describe('NotAllowedPage component', () => {
  afterEach(cleanup);

  it('matches snapshot', () => {
    const { asFragment } = renderWithRouter(<NotAllowedPage />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders', () => {
    renderWithRouter(<NotAllowedPage />);
  });
});
