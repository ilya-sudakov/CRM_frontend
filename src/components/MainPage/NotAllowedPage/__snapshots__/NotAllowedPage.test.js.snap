// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotAllowedPage component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="error-page"
  >
    <div
      class="error-page__title"
    >
      <span
        class="error-page__error-code"
      >
        403
      </span>
      Доступ на эту страницу запрещен
    </div>
    <div
      class="error-page__description"
    >
      Вы не имеете доступа к этой странице.
      <br />
      <br />
      Если это ошибка, и запрашиваемая страница должна быть доступна для просмотра - напишите об этом в 
      <a
        class="error-page__link"
        href="/feedback"
      >
        обратной связи
      </a>
    </div>
    <a
      class="error-page__button"
      href="/"
    >
      На главную
    </a>
  </div>
</DocumentFragment>
`;
